src/modules/constructRelationalGraph.js

```diff
@@ -0,0 +1,265 @@
+// @flow
+import {original} from 'immer';
+import Doctor from '../resources/Doctor.json';
+import Patient from '../resources/Patient.json';
+import Department from '../resources/Department.json';
+
+export type Expression = {[key: string]: any};
+
+export type JoinCondition = {[key: string]: any};
+
+export type SelectionCondition = {[key: string]: any};
+
+export type Graph = {
+  [key: string]: {
+    edges: {[key: string]: Array<JoinCondition>},
+    selections: Array<SelectionCondition>,
+  },
+};
+
+export const AvailableTables = ['Doctor', 'Patient', 'Department'];
+
+export const ExpressionType = {
+  SELECTION: 'selection',
+  JOIN: 'join',
+  RELATION: 'relation',
+  AND: 'and',
+  CMP: 'cmp',
+};
```

Comment:
It isn't necessarily a bad idea to use constants for these, but given that the rest of the code does not, I would suggest removing this.